3648/2/commands/start.rb

```diff
@@ -0,0 +1,48 @@
+require 'redis'
+require_relative 'help_methods'
+
+module StartCommand
+  include HelpMethods
+  def start!(*)
+    if registred?
+      respond_with :message, text: "Вы уже вошли под: #{session[:number]}"
+    else
+      save_context :register_from_message
+      respond_with :message, text: 'Введите ваш номер:'
+    end
+  end
+
+  def register_from_message(*words)
+    respond_with :message, text: register_session(words)
+  end
+
+  private
+
+  def register_session(number = nil, *)
+    redis = Redis.new
+    check_registaration(redis, number.first.to_i)
+  rescue NoMethodError
+    save_context :register_from_message
+    respond_with :message, text: 'Что-то пошло не так, попробуйте еще раз:'
+  end
+
+  def check_registaration(redis, number)
+    if redis.get(number) || session.key?(:number)
+      'Вы уже зарегестрированны'
+    else
+      return 'Данный номер отсутсвует. Попробуйте еще раз' unless member?(number)
```

Comment:
You can use `exclude?` inside `member?` method and make this line and option of `elsif`